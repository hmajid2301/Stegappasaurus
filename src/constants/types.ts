export type ThemeColors = '#17212d' | '#ffffff';
export type TabColors = '#009cff' | '#e88c0c';

export interface Theme {
  background: ThemeColors;
  color: ThemeColors;
  isDark: boolean;
}
