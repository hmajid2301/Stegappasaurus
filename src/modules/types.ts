export type ThemeColors = "#17212D" | "#FFF";
export type PrimaryColor = "#009CFF" | "#E88C0C";

export interface ITheme {
  background: ThemeColors;
  color: ThemeColors;
  isDark: boolean;
}
