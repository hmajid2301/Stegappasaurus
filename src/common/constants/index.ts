import {
  BackgroundColors,
  ITheme,
  PrimaryColor,
  PrimaryColorNames,
  ThemeColors
} from "~/common/interfaces";
import { colors } from "~/common/styles";

interface IThemes {
  DARK_THEME: ITheme;
  LIGHT_THEME: ITheme;
}

interface IPrimaryColor {
  BLUE: {
    name: PrimaryColorNames;
    color: PrimaryColor;
  };
  ORANGE: {
    name: PrimaryColorNames;
    color: PrimaryColor;
  };
}

export const PRIMARY_COLORS: IPrimaryColor = {
  BLUE: { name: "BLUE", color: colors.primary as PrimaryColor },
  ORANGE: { name: "ORANGE", color: colors.secondary as PrimaryColor }
};

export const THEMES: IThemes = {
  DARK_THEME: {
    background: colors.darkColor as BackgroundColors,
    color: colors.pureWhite as ThemeColors,
    isDark: true
  },

  LIGHT_THEME: {
    background: colors.pureWhite as BackgroundColors,
    color: colors.pureBlack as ThemeColors,
    isDark: false
  }
};
