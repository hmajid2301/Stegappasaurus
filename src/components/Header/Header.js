import PropTypes from 'prop-types';
import React from 'react';
import { Header, Icon } from 'react-native-elements';

import Logo from '../Logo';
import styles from './styles';


const CustomHeader = ({ navigation, color }) => (
  <Header
    leftComponent={<Icon name='menu' type='simple-line-icon' color={styles(color).icons.color} onPress={() => navigation.openDrawer()}/>}
    centerComponent={<Logo />}
    outerContainerStyles={styles(color).outerHeaderContainer}
    innerContainerStyles={styles(color).innerHeaderContainer}
  />
);

CustomHeader.propTypes = {
  navigation: PropTypes.shape({
    navigate: PropTypes.func.isRequired,
  }).isRequired,
  color: PropTypes.string.isRequired,
};

export default CustomHeader;
