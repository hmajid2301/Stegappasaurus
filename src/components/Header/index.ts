import AppHeader from './AppHeader';
import MainHeader from './MainHeader';

export {AppHeader, MainHeader};
