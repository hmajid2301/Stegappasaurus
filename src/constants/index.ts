import {
  BackgroundColors,
  ITheme,
  PrimaryColor,
  PrimaryColorNames,
  ThemeColors
} from "@types";

interface IThemes {
  DARK_THEME: ITheme;
  LIGHT_THEME: ITheme;
}

interface IPrimaryColor {
  BLUE: {
    name: PrimaryColorNames;
    color: PrimaryColor;
  };
  ORANGE: {
    name: PrimaryColorNames;
    color: PrimaryColor;
  };
}

export const fonts = {
  body: "Roboto",
  bodyLight: "Roboto-Light",
  header: "Roboto-Thin"
};

export const colors = {
  darkColor: "#17212D",
  iconBlack: "#222",
  iconBlue: "#007CFE",
  iconGreen: "#4CDA64",
  iconLightGreen: "#25AE88",
  iconOrange: "#FF9501",
  iconRed: "#FD3C2D",
  primary: "#009CFF",
  pureBlack: "#000",
  pureWhite: "#FFF",
  secondary: "#E88C0C"
};

export const PRIMARY_COLORS: IPrimaryColor = {
  BLUE: { name: "BLUE", color: colors.primary as PrimaryColor },
  ORANGE: { name: "ORANGE", color: colors.secondary as PrimaryColor }
};

export const THEMES: IThemes = {
  DARK_THEME: {
    background: colors.darkColor as BackgroundColors,
    color: colors.pureWhite as ThemeColors,
    isDark: true
  },

  LIGHT_THEME: {
    background: colors.pureWhite as BackgroundColors,
    color: colors.pureBlack as ThemeColors,
    isDark: false
  }
};
