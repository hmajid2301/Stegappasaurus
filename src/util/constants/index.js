import { colors } from '../styles';


export const PRIMARY_COLORS = {
  blue: { name: 'blue', hexCode: colors.primary },
  orange: { name: 'orange', hexCode: colors.secondary },
};

export const THEMES = {
  LIGHT_THEME: { dark: false, backgroundColor: colors.pureWhite, color: colors.pureBlack },
  DARK_THEME: { dark: true, backgroundColor: colors.darkColor, color: colors.pureWhite },
};
