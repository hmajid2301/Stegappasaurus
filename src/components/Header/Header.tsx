import React from 'react';
import {StatusBar, StyleProp, View, ViewStyle} from 'react-native';
import {
  Header as ElementsHeader,
  HeaderSubComponent,
} from 'react-native-elements';
import {NavigationScreenProp} from 'react-navigation';

import {ITheme, TabColors} from '~/constants/types';
import styles from './styles';

interface IProps {
  center?: HeaderSubComponent;
  navigation: NavigationScreenProp<any, any>;
  left?: HeaderSubComponent;
  primary: TabColors;
  right?: HeaderSubComponent;
  styles?: StyleProp<ViewStyle>;
  theme: ITheme;
}

export default class Header extends React.Component<IProps, {}> {
  public render() {
    const {primary, theme} = this.props;
    return (
      <View>
        <StatusBar
          backgroundColor={theme.background}
          hidden={false}
          barStyle={theme.isDark ? 'light-content' : 'dark-content'}
        />
        <ElementsHeader
          centerComponent={this.props.center}
          containerStyle={[
            styles.container,
            {
              backgroundColor: theme.background as string,
              borderBottomColor: primary,
            },
            this.props.styles,
          ]}
          leftComponent={this.props.left}
          rightComponent={this.props.right}
        />
      </View>
    );
  }
}
