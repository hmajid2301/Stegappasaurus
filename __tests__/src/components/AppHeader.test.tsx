import {render, fireEvent} from '@testing-library/react-native';
import React from 'react';

import {AppHeader} from '~/components/Header';
import {ITheme} from '~/constants/types';

const LIGHT_THEME: ITheme = {
  background: '#ffffff',
  color: '#17212d',
  isDark: false,
};

const DARK_THEME: ITheme = {
  background: '#17212d',
  color: '#ffffff',
  isDark: false,
};

const navigation: any = {
  navigate: jest.fn(),
};
describe('AppHeader: Functionality', () => {
  test('Press home icon', () => {
    const {getByTestId} = render(
      <AppHeader
        navigation={navigation}
        primary="#009cff"
        theme={LIGHT_THEME}
      />,
    );

    const spy = jest.spyOn(navigation, 'navigate');
    const homeIcon = getByTestId('home');
    fireEvent.press(homeIcon);
    expect(spy).toHaveBeenCalledWith('Main');
  });

  test('Press home icon: No navigation props', () => {
    const {getByTestId} = render(
      <AppHeader navigation={{} as any} primary="#009cff" theme={DARK_THEME} />,
    );

    const homeIcon = getByTestId('home');
    try {
      fireEvent.press(homeIcon);
    } catch (error) {
      expect(error).toBeTruthy();
    }
  });
});
