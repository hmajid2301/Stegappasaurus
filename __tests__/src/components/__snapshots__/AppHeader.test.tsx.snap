// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader: Match snapshots 1 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#17212D"
    barStyle="dark-content"
    hidden={false}
    showHideTransition="fade"
  />
  <Themed.Header
    centerComponent={
      <Logo
        color="#FFF"
        isDark={false}
      />
    }
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#17212D",
          "borderBottomColor": "#009CFF",
        },
      ]
    }
    leftComponent={
      <Themed.Icon
        color="#FFF"
        name="menu"
        onPress={[Function]}
        type="simple-line-icon"
      />
    }
    rightComponent={
      <Themed.Icon
        color="#FFF"
        name="home"
        onPress={[Function]}
        type="antdesign"
      />
    }
  />
</View>
`;

exports[`AppHeader: Match snapshots 2 1`] = `
<View>
  <StatusBar
    animated={false}
    backgroundColor="#FFF"
    barStyle="light-content"
    hidden={false}
    showHideTransition="fade"
  />
  <Themed.Header
    centerComponent={
      <Logo
        color="#17212D"
        isDark={true}
      />
    }
    containerStyle={
      Array [
        Object {
          "borderBottomWidth": 2,
        },
        Object {
          "backgroundColor": "#FFF",
          "borderBottomColor": "#009CFF",
        },
      ]
    }
    leftComponent={
      <Themed.Icon
        color="#17212D"
        name="menu"
        onPress={[Function]}
        type="simple-line-icon"
      />
    }
    rightComponent={
      <Themed.Icon
        color="#17212D"
        name="home"
        onPress={[Function]}
        type="antdesign"
      />
    }
  />
</View>
`;
